spring-cloud-sleuth-zipkin-stream/src/main/java/org/springframework/cloud/sleuth/zipkin/stream/SamplingZipkinSpanIterator.java

```diff
@@ -110,17 +130,17 @@ public void remove() {
 		// A zipkin span without any annotations cannot be queried, add special "lc" to
 		// avoid that.
 		if (span.logs().isEmpty() && span.tags().isEmpty()) {
-			String processId = span.getProcessId() != null
-					? span.getProcessId().toLowerCase()
-					: ZipkinMessageListener.UNKNOWN_PROCESS_ID;
-			zipkinSpan.addBinaryAnnotation(
-					BinaryAnnotation.create(Constants.LOCAL_COMPONENT, processId, ep));
+			addLocalComponentAnnotation(span, zipkinSpan, ep);
 		}
 		else {
 			ZipkinMessageListener.addZipkinAnnotations(zipkinSpan, span, ep);
 			ZipkinMessageListener.addZipkinBinaryAnnotations(zipkinSpan, span, ep);
 		}
-
+		if (!spanContainsAnyZipkinConstant(span)) {
+			addLocalComponentAnnotation(span, zipkinSpan, ep);
+		} else if (spanContainsAnyZipkinClientConstantAndSaIsNotSet(span)) {
```

Comment:
add a comment here why we are adding the "server address".

The guidance from the docs is "When logging CLIENT_SEND, instrumentation should also log the SERVER_ADDR", so for example you can link to that: https://github.com/openzipkin/zipkin-java/blob/master/zipkin/src/main/java/zipkin/Constants.java#L28